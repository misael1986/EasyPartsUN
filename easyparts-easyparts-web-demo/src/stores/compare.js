import { decorate, observable } from "mobx";

class Compare {
  constructor() {
    this.parts = [];
  }
}

decorate(Compare, {
  parts: observable,
});

export default new Compare();
