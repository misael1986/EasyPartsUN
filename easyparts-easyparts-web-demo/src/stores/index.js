export { default as SearchStore } from "./search";
export { default as UserStore } from "./user";
export { default as CompareStore } from "./compare";
export { default as PurposeStore } from "./purpose";
