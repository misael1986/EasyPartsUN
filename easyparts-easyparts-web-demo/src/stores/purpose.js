import { decorate, observable } from "mobx";

class Purpose {
  constructor() {
    this.name = null;
    this.list = [];
  }
}

decorate(Purpose, {
  name: observable,
  list: observable,
});

export default new Purpose();
