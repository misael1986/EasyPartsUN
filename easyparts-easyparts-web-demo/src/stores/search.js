import { decorate, observable } from "mobx";

class Search {
  constructor() {
    this.cpuList = [];
    this.motherboardList = [];
    this.gpuList = [];
    this.ramList = [];
    this.storage1List = [];
    this.storage2List = [];
    this.chassisList = [];
    this.powerSourceList = [];

    this.typeSelected = "cpu";

    this.searchList = [];
    this.searchText = "";
    this.componentDescription = "Procesador";

    this.cpuFilters = [];
    this.motherboardFilters = [];
    this.gpuFilters = [];
    this.ramFilters = [];
    this.storage1Filters = [];
    this.storage2Filters = [];
    this.chassisFilters = [];
    this.powerSourceFilters = [];
  }

  clear() {
    this.cpuList = [];
    this.motherboardList = [];
    this.gpuList = [];
    this.ramList = [];
    this.storage1List = [];
    this.storage2List = [];
    this.chassisList = [];
    this.powerSourceList = [];

    this.typeSelected = "cpu";

    this.searchList = [];
    this.searchText = "";
    this.componentDescription = "Procesador";

    this.cpuFilters = [];
    this.motherboardFilters = [];
    this.gpuFilters = [];
    this.ramFilters = [];
    this.storage1Filters = [];
    this.storage2Filters = [];
    this.chassisFilters = [];
    this.powerSourceFilters = [];
  }

  get filterFromType() {
    return this[this.typeSelected + "Filters"];
  }

  get searchListFiltered() {
    let byName = this.searchList.filter((it) => {
      return it.nombre.includes(this.searchText);
    });
    const filtersForType = this.filterFromType;
    for (let i = 0; i < filtersForType.length; i++) {
      byName = byName.filter((it) => {
        return !(
          filtersForType[i].selected !== null &&
          it[filtersForType[i].databaseName] !== filtersForType[i].selected
        );
      });
    }
    return byName;
  }
}

decorate(Search, {
  searchList: observable,
  searchText: observable,

  componentDescription: observable,

  cpuFilters: observable,
  motherboardFilters: observable,
  gpuFilters: observable,
  ramFilters: observable,
  storage1Filters: observable,
  storage2Filters: observable,
  chassisFilters: observable,
  powerSourceFilters: observable,
});

export default new Search();
