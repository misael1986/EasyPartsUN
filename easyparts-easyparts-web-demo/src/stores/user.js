import { computed, decorate, observable } from "mobx";

class User {
  constructor() {
    this.email = "";
    this.password = "";
    this.otp = "";
    this.loggedIn = false;

    this.pc = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities = {
      ram: 0,
      storage1: 0,
      storage2: 0,
      gpu: 0,
    };

    this.pc2 = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities2 = null;

    this.routeOrigin = "/";
  }

  setRamQuantity(increase) {
    if (increase && this.quantities.ram === 4) return;
    if (!increase && this.quantities.ram === 1) return;

    this.quantities = {
      ...this.quantities,
      ram: (this.quantities.ram = increase
        ? this.quantities.ram * 2
        : this.quantities.ram / 2),
    };
  }

  setStorageQuantity(id, increase) {
    const prev = this.quantities["storage" + id];
    if (increase && prev > 2) return;
    if (!increase && prev === 1) return;

    if (increase && this.pc["storage" + id].tecnologia === "M2") return;

    this.quantities = {
      ...this.quantities,
      ["storage" + id]: (this.quantities["storage" + id] += increase ? 1 : -1),
    };
  }

  get pcPower() {
    let total = 0;
    Object.keys(this.pc).forEach((it) => {
      if (this.pc[it] !== null && this.pc[it].potencia_max !== undefined) {
        total += this.pc[it].potencia_max;
      }
    });
    return total;
  }

  getPcPrice(pc, quantities) {
    let total = 0;
    Object.keys(pc).forEach((it) => {
      if (pc[it] !== null && pc[it].precio !== undefined) {
        if (quantities[pc[it].type] !== undefined) {
          total += pc[it].precio * quantities[pc[it].type];
        } else {
          total += pc[it].precio;
        }
      }
      //console.log(toJS(pc[it]));
      //console.log("New total with type: " + pc[it].type + " = " + total);
    });
    return total;
  }

  clear() {
    this.email = "";
    this.password = "";
    this.otp = "";
    this.loggedIn = false;

    this.clearPc();
    this.clearPc2();

    this.routeOrigin = "/";
  }

  clearPc() {
    this.pc = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities = {
      ram: 0,
      storage1: 0,
      storage2: 0,
      gpu: 0,
    };
  }

  clearPc2() {
    this.pc2 = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities2 = {
      ram: 0,
      storage1: 0,
      storage2: 0,
      gpu: 0,
    };
  }
}

decorate(User, {
  email: observable,
  password: observable,
  otp: observable,
  loggedIn: observable,

  pc: observable,
  pcPower: computed,

  quantities: observable,
});

export default new User();
