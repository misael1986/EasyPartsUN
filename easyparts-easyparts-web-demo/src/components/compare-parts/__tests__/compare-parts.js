import React from "react";
import renderer from "react-test-renderer";
import CompareParts from "../compare-parts";
import { Provider } from "mobx-react";
import { CompareStore } from "../../../stores";

import "mobx-react-lite/batchingForReactDom";

describe("CompareParts", () => {
  it("hasn't changed unexpectedly", () => {
    const tree = renderer
      .create(
        <Provider compareStore={CompareStore}>
          <CompareParts />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
