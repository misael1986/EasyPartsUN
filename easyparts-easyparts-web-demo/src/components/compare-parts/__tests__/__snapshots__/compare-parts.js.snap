// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareParts hasn't changed unexpectedly 1`] = `null`;
